subsys/net/ip/net_if.c

```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
 	return NULL;
 }
 
+int net_if_get_name(struct net_if *iface, char *buf, int len)
+{
+#if defined(CONFIG_NET_INTERFACE_NAME)
+	int name_len;
+
+	if (iface == NULL || buf == NULL || len <= 0) {
+		return -EINVAL;
+	}
+
+	name_len = strlen(net_if_get_config(iface)->name);
+	if (name_len > len) {
+		return -ERANGE;
+	}
+
+	strncpy(buf, net_if_get_config(iface)->name, name_len);
+
+	return name_len;
+#else
+	return -ENOTSUP;
+#endif
+}
+
+int net_if_set_name(struct net_if *iface, char *buf)
+{
+#if defined(CONFIG_NET_INTERFACE_NAME)
+	int name_len;
+
+	if (iface == NULL || buf == NULL) {
+		return -EINVAL;
+	}
+
+	name_len = strlen(buf);
+	if (name_len > CONFIG_NET_INTERFACE_NAME_LEN) {
+		return -ENAMETOOLONG;
+	}
+
+	memset(net_if_get_config(iface)->name, 0,
+	       CONFIG_NET_INTERFACE_NAME_LEN + 1);
+	strncpy(net_if_get_config(iface)->name, buf, name_len);
```

Comment:
Same here, `name_len` does not count NULL terminator, so it'd be omitted (although in this particular case memset above guarantees the NULL terminator will be in the buffer anyway). I think `sizeof(net_if_get_config(iface)->name)` would be more appropriate.